http.go

```diff
@@ -0,0 +1,311 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"strconv"
+	"time"
+)
+
+type respTasks struct {
+	Tasks []Task `json:"tasks"`
+}
+
+type respOkErr struct {
+	Id    string `json:"id,omitempty"`
+	Error string `json:"error,omitempty"`
+}
+
+// handlers
+func handleNextDate(w http.ResponseWriter, r *http.Request) {
+
+	reqNow := r.FormValue("now")
+	reqDate := r.FormValue("date")
+	reqRepeat := r.FormValue("repeat")
+
+	now, err := parseDate(reqNow)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	nextDate, err := NextDate(now, reqDate, reqRepeat)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	fmt.Fprint(w, nextDate)
+}
+
+func handleTask(w http.ResponseWriter, r *http.Request) {
+
+	switch r.Method {
+	case http.MethodPost:
+		handleTaskPOST(w, r)
+	case http.MethodGet:
+		handleTaskGET(w, r)
+	case http.MethodPut:
+		handleTaskPUT(w, r)
+	case http.MethodDelete:
+		handleTaskDELETE(w, r)
+	default:
+		w.WriteHeader(http.StatusMethodNotAllowed)
+	}
+}
+
+func handleTaskPOST(w http.ResponseWriter, r *http.Request) {
+
+	var respBody []byte
+	var err error
+
+	id, err := addTask(r.Body)
+	if err != nil {
+		respBody, err = json.Marshal(respOkErr{Error: err.Error()})
+	} else {
+		respBody, err = json.Marshal(respOkErr{Id: strconv.Itoa(id)})
+	}
+
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	w.Write(respBody)
```

Comment:
Вообще в идеале ошибку записи ответа логировать, она иногда помогает понять почему клиент не получил ответ
```suggestion
	_, err = w.Write(respBody)
	if err != nil {
		fmt.Println("Error in writing response", err.Error())
	}
```